templatetags/htk_tags.py

```diff
@@ -219,6 +220,18 @@ def get_request_duration():
     duration = timer.duration()
     return duration
 
+@register.simple_tag(takes_context=True)
+def has_permission(context, codename):
+    request_meta_data = context.get('request', None)
+    request = request_meta_data.get('request', None)
+    user = request.user
```

Comment:
add a check for `if request` and/or `if user` (defensive coding)

if not, `has_permission = False`